src/compiler/sys/logger/test/terminal-logger.spec.ts

```diff
@@ -0,0 +1,209 @@
+      ['debug', true],
+      ['info', true],
+      ['warn', false],
+      ['error', false],
+    ])("shouldLog for info at level '%s' should be '%p'", (currentLevel, expected) => {
+      expect(shouldLog(currentLevel, 'info')).toBe(expected);
+    });
+
+    it.each<[LogLevel, boolean]>([
+      ['debug', true],
+      ['info', false],
+      ['warn', false],
+      ['error', false],
+    ])("shouldLog for debug messages at level '%s' should be '%p'", (currentLevel, expected) => {
+      expect(shouldLog(currentLevel, 'debug')).toBe(expected);
+    });
+  });
+
+  describe('basic logging functionality', () => {
+    const originalLog = console.log;
+    const originalWarn = console.warn;
+    const originalError = console.error;
+
+    afterAll(() => {
+      console.log = originalLog;
+      console.warn = originalWarn;
+      console.error = originalError;
+    });
+
+    function setup() {
+      const logMock = jest.fn();
+      const warnMock = jest.fn();
+      const errorMock = jest.fn();
+
+      console.log = logMock;
+      console.warn = warnMock;
+      console.error = errorMock;
+
+      const loggerSys = createNodeLoggerSys(process);
+
+      loggerSys.memoryUsage = jest.fn().mockReturnValue(10_000_000);
+
+      const writeLogsMock = jest.fn();
+      loggerSys.writeLogs = writeLogsMock;
+
+      const logger = createTerminalLogger(loggerSys);
+
+      jest.useFakeTimers().setSystemTime(new Date(2022, 6, 3, 9, 32, 32, 32));
+
+      return { logger, logMock, warnMock, errorMock, writeLogsMock };
+    }
+
+    it("supports 'debug' level", () => {
+      const { logger, logMock } = setup();
+      logger.setLevel('debug');
+      logger.debug('my debug message');
+      expect(logMock).toBeCalledWith(`${cyan('[32:32.0]')}  my debug message ${dim(' MEM: 10.0MB')}`);
+    });
+
+    it("supports 'info' level", () => {
+      const { logger, logMock } = setup();
+      logger.info('my info message');
+      expect(logMock).toBeCalledWith(`${dim('[32:32.0]')}  my info message`);
+    });
+
+    it("supports 'warn' level", () => {
+      const { logger, warnMock } = setup();
+      logger.warn('my warn message');
+      expect(warnMock).toBeCalledWith(`\n${bold(yellow('[ WARN  ]'))}  my warn message\n`);
+    });
+
+    it("supports 'error' level", () => {
+      const { logger, errorMock } = setup();
+      logger.error('my error message');
+      expect(errorMock).toBeCalledWith(`\n${bold(red('[ ERROR ]'))}  my error message\n`);
+    });
+
+    it('supports shipping logs to a file', function () {
+      const { logger, writeLogsMock } = setup();
+      logger.setLogFilePath!('testfile.txt');
+      logger.info('test message');
+      logger.writeLogs!(false);
+      const expectedLogfile = [
+        '09:32:32.00  0010.0MB  I  test message',
+        '09:32:32.00  0010.0MB  F  --------------------------------------',
+      ].join('\n');
+      expect(writeLogsMock).toBeCalledWith('testfile.txt', expectedLogfile, false);
+    });
```

Comment:
Can we add a small test that we don't write to a file if the log file path isn't set?

```suggestion
    });
    
    it("doesn't write to any logs if the log path isn't set", function () {
      const { logger, writeLogsMock } = setup();
      logger.setLogFilePath!('');
      logger.info('test message');
      logger.writeLogs!(false);
      expect(writeLogsMock).not.toHaveBeenCalled();
    });
```